content/spin/plugin-authoring.md

```diff
@@ -0,0 +1,87 @@
+title = "Creating Spin Plugins"
+template = "spin_main"
+date = "2023-01-26T00:22:56Z"
+---
+
+Spin plugins add new functionality or subcommands to Spin without modifying the
+Spin codebase. They make Spin easily extensible while keeping it lightweight.
+Spin plugins can add new triggers to Spin (such as the [example timer
+trigger](https://github.com/fermyon/spin/blob/main/examples/spin-timer/trigger-timer.json)),
+enable new language SDKs (such as
+[`js2wasm`](https://github.com/fermyon/spin-plugins/blob/main/manifests/js2wasm/js2wasm.json)),
+and more.
+
+This document will cover what Spin plugins are, how to use a plugin, and how to
+contribute a plugin.
+
+## What are Spin Plugins?
+
+A Spin plugin is an executable that is added to Spin's plugins directory
+(`$XDG_DATA_HOME/spin/plugins`) upon a `spin plugins install <plugin-name>`. The
+plugin is then ready to be used. If the plugin is an extension to the Spin CLI,
+it can now be executed directly as a subcommand: `spin <plugin-name>`. If the
+plugin is a trigger plugin, it will be executed during `spin up` when an app
+using that trigger is run. 
+
+While for now plugins are assumed to be executables, in the future, support for
+plugging in WebAssembly modules may be desirable.
+
+## How to Find and Use a Spin Plugin
+
+Spin maintains a [Spin plugins
+repository](https://github.com/fermyon/spin-plugins) that acts as an inventory
+of available plugins. Use the `spin plugins` command to list plugins (available
+and installed) and install a new plugin. For example, the `js2wasm` plugin is
+needed in order to write JavaScript Spin applications. It can be installed from
+the Spin plugins repository, using the following command: 
+
+```sh
+spin plugins install js2wasm
+```
+
+With the plugin installed, you can call `spin js2wasm` as part of building a
+JavaScript component; for example, in the component build directive. Learn more
+about building Spin components in JavaScript
+[here](https://developer.fermyon.com/spin/javascript-components.md).
+
+## Authoring a Spin Plugin
+
+As mentioned previously, Spin plugins are executables that are executed either
+directly via the Spin CLI or as a trigger. 
+
+Creating and using a plugin can be summarized as follows:
+1. Build, package, and zip the plugin
+2. Create a Spin plugin manifest for the plugin
+3. Install the plugin (by pointing to the plugin manifest with the `--file` or
+   `--url` flag)
+
+Plugins are delivered as `.tar.gz` files, so after creating the executable, you
+must create a Spin plugin manifest to define it. A Spin plugin manifest is a
+JSON file that conforms to the [Spin plugin manifest
+schema](https://github.com/fermyon/spin-plugins/blob/main/json-schema/spin-plugin-manifest-schema-0.1.json).
+A plugin manifest defines a plugin’s name, version, license, homepage (i.e.
+GitHub repo), compatible Spin version, and gives a short description of the
+plugin. It also points to the url (whether `file://` or network such as
```

Comment:
Suggest "URLs ... of the tar archives ..." (to be clear that there might be multiple packages)